File: doc/home/eg_bars.m
qfigure('eg_bars', 3, 2.5);

yy = [
  1.3 1.5
  0.6 0.7
  1.25 1.4
  ];

qlegopt x0 3.6 y0 1.5 

qbrush 955
qpen none
qbars([1:3]-.2, yy(:,1), .3);
qplegend 'Trial A'

qbrush 559
qbars([1:3]+.2, yy(:,2), .3);
qplegend 'Trial B'

qpen k 0 solid
qxaxis(0, [.5 3.5], [1:3], {'Baseline', 'Expt.', 'Wash'});
qyaxis(.5, [0:.5:1.5], 'Activity');

ym = max(yy,[],2);
dy = [ nan 5 30];
txt = {'', '**', 'n.s.'};
for k=2:3
  qshiftedline(1+.35*[-1 1], ym(1), 0, -dy(k));
  qshiftedline(k+.35*[-1 1], ym(k), 0, -dy(k));
  qshiftedline([1 1 k k], [ym(1) ym(1) ym(1) ym(k)], 0, -dy(k) - [0 10 10 0]);
  qat(mean([1 k]), max(ym([1 k])));
  qalign bottom center
  qtext(0, -dy(k)-12, txt{k});
end

qshrink

qsave pdf
qsave png 120

File: doc/home/eg_bars.py
#!/usr/bin/python3

import numpy as np
import qplot as qp

qp.figure('eg_bars', 3, 2.5)

yy = np.array([[1.3, 1.5], [0.6, 0.7], [1.25, 1.4]])

qp.legopt(x0=2.6, y0=1.5)

qp.brush('955')
qp.pen('none')
qp.bars(np.arange(3) - .2, yy[:,0], .3)
qp.plegend('Trial A')

qp.brush('559')
qp.bars(np.arange(3) + .2, yy[:,1], .3)
qp.plegend('Trial B')

qp.pen('k', 0)
qp.xcaxis('', np.arange(3), ['Baseline', 'Expt.', 'Wash'])
qp.yaxis('Activity', qp.arange(0, 1.5, .5), x=-.5)

ym = np.max(yy, 1)
dy = [ np.nan, 5, 30]
txt = ['', '**', 'n.s.']
for k in [1, 2]:
    qp.shiftedline(0+.35*np.array([-1, 1]), ym[0], 0, -dy[k])
    qp.shiftedline(k+.35*np.array([-1, 1]), ym[k], 0, -dy[k])
    qp.shiftedline(np.array([0, 0, k, k]), ym[[0,0,0,k]],
                   0, -dy[k] - np.array([0, 10, 10, 0]))
    qp.at(np.mean([0, k]), np.max(ym[[0,k]]))
    qp.align('center', 'bottom')
    qp.text(txt[k], dy=-dy[k]-12)
    
qp.shrink()

qp.save('pdf')

